http/auth-gateway.js

```diff
@@ -17,6 +19,44 @@ var HttpAuthGateway = HttpGateway.extend({
         }
 
         return options;
+    },
+
+    /**
+     * Handle 401 Unauthorized responses
+     *
+     * Assume that the oauth access token has expired and the refresh token
+     * needs to be exchanged for a new one.
+     */
+    handle401 : function(resolve, reject, method, path, data, headers)
+    {
+        var gateway, token, refreshData, refreshHeaders, handleSuccess, handleFailure;
+
+        gateway = this;
+        token   = store.get('token');
+
+        data = data || {};
+
+        handleSuccess = function (accessToken) {
```

Comment:
What do the `token` and `accessToken` objects look like? `accessToken` is really the response data, right? I'm a little confused by these being objects instead of strings.